device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala

```diff
@@ -50,41 +57,43 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace])
   def listGroups(ns: Namespace): Route =
     complete(db.run(GroupInfoRepository.list(ns)))
 
-  def fetchGroupInfo(groupId: Uuid, ns: Namespace): Route =
+  def fetchGroupInfo(groupId: Uuid): Route =
     complete(db.run(GroupInfoRepository.getGroupInfoById(groupId)))
 
   def createGroupInfo(id: Uuid, groupName: Name, namespace: Namespace, data: Json): Route =
     complete(Created -> db.run(GroupInfoRepository.create(id, groupName, namespace, data)))
 
-  def updateGroupInfo(groupId: Uuid, groupName: Name, namespace: Namespace, data: Json): Route =
+  def updateGroupInfo(groupId: Uuid, groupName: Name, data: Json): Route =
     complete(db.run(GroupInfoRepository.updateGroupInfo(groupId, data)))
 
   def renameGroup(groupId: Uuid, newGroupName: Name): Route =
     complete(db.run(GroupInfoRepository.renameGroup(groupId, newGroupName)))
 
   val route: Route =
-    (pathPrefix("device_groups") & namespaceExtractor) { ns =>
-      (post & path("from_attributes")) {
-        entity(as[CreateGroupRequest]) { groupInfo => createGroupFromDevices(groupInfo, ns) }
-      } ~
-      (get & pathEnd) {
-        listGroups(ns)
+    pathPrefix("device_groups") {
+      namespaceExtractor { ns =>
+        (post & path("from_attributes")) {
+          entity(as[CreateGroupRequest]) { groupInfo => createGroupFromDevices(groupInfo, ns) }
+        } ~
+        (post & extractUuid & pathEnd & parameter('groupName.as[Name])) { (groupId, groupName) =>
+          entity(as[Json]) { body => createGroupInfo(groupId, groupName, ns, body) }
+        } ~
+        (get & pathEnd) {
+          listGroups(ns)
+        }
       } ~
-      extractUuid { groupId =>
+      groupId { groupId =>
```

Comment:
Shadowing that could be confusing, rename first groupId to groupIdExtractor to be more consistent with other code.